src/lib/y2storage/blk_device.rb

```diff
@@ -82,6 +82,36 @@ def blk_filesystem
       encrypted? ? encryption.direct_blk_filesystem : direct_blk_filesystem
     end
 
+    alias_method :filesystem, :blk_filesystem
+
+    # Label of the filesystem, if any
+    # @return [String, nil]
+    def filesystem_label
+      return nil unless blk_filesystem
+      blk_filesystem.label
+    end
+
+    # UUID of the filesystem, if any
+    # @return [String, nil]
+    def filesystem_uuid
+      return nil unless blk_filesystem
+      blk_filesystem.uuid
+    end
+
+    # Type of the filesystem, if any
+    # @return [Filesystems::Type, nil]
+    def filesystem_type
+      return nil unless blk_filesystem
+      blk_filesystem.type
+    end
+
+    # Mount point of the filesystem, if any
+    # @return [String, nil]
+    def filesystem_mountpoint
+      return nil unless blk_filesystem
+      blk_filesystem.mountpoint
+    end
```

Comment:
well, from my POV what is wrong, that it creates high coupling between blk_device and filesystem. Previously blk_device responsibility was to return filesystem or nil if there is no such filesystem. Now it depends on their API. so now blk_device have to know that filesystem have given subset of methods. see e.g. https://thebojan.ninja/2015/04/08/high-cohesion-loose-coupling/ coupling part.

consider e.g. that method mountpoint get argument. in this case you have to change filesystem method, its caller and additionally also this method.